trlx/data/configs.py

```diff
@@ -1,20 +1,22 @@
+from copy import deepcopy
 from dataclasses import dataclass, field
-from typing import Any, Dict, Optional, Set
+from typing import Any, Dict, Optional
 
 import yaml
 
 from trlx.data.method_configs import MethodConfig, get_method
 
 
-def merge(base: Dict, update: Dict, updated: Set) -> Dict:
+def merge(base: Dict, update: Dict) -> Dict:
```

Comment:
Just a note: the sweep utils are broken because the sweep configs provide a single key and I believe this and the previous implementation check the whole key-path for merging. E.g. you can't do the following merge:  
```python
merge({'optimizer': 'kwargs': 'lr': 1e-3}, {'lr': 9}) != {'optimizer': 'kwargs': 'lr': 9}
```
https://colab.research.google.com/drive/1xEhbdJjuw40ow1ObhiWmAfD1wt2R6zb2?usp=sharing

Probably best to handle this from the sweep side with a manual `evolve`. CC @reciprocated who's probably aware of this but thought I'd tag if otherwise.

Reply:
I've updated it to be of the following form: https://github.com/CarperAI/trlx/blob/deb667a00cb66a1dd20f9a61ea6d7482637ac6e3/configs/sweeps/ppo_sweep.yml#L9-L14
https://github.com/CarperAI/trlx/blob/deb667a00cb66a1dd20f9a61ea6d7482637ac6e3/trlx/data/configs.py#L275-L290

Reply:
> side with a manual `evolve`. CC @reciprocated who's probably aware of this but thought I'd tag if otherwise.

Oh, I thought was actually a bug rather than feature. I see. I suppose if this is to be changed/fixed in @reciprocated 's PR I'll make a new merge function for the current use